### ISSUE TITLE
setup.py imports from Cython before checking for Cython

### ISSUE THREAD

**Issue by jsolbrig** [Unidata/cftime#34]
As discussed in Unidata/netcdf4-python#767, setup.py calls `from Cython.Build import cythonize` before it requires Cython to be installed.  As a consequence, building without Cython previously installed fails with the following error:

```
Traceback (most recent call last):
  File "setup.py", line 3, in <module>
    from Cython.Build import cythonize
ImportError: No module named Cython.Build
```

**Comment by jswhit** [comment]
I think [this](pull request #35) should fix it (untested).  Based on https://stackoverflow.com/questions/37471313/setup-requires-with-cython

**Comment by jswhit** [comment]
@jsolbrig, can you test this and let us know if it fixes the problem?

**Comment by jsolbrig** [comment]
@jswhit Yep, that solved it.  I'm also able to build netcdf4-python against it.

**Comment by jhamman** [comment]
closed via #35

**Comment by terokinnunen** [comment]
Looks like this issue came back in release 1.0.3.
```
[terokinnunen@localhost tmp]$ python3 -m venv venv
[terokinnunen@localhost tmp]$ source venv/bin/activate
(venv) [terokinnunen@localhost tmp]$ pip install -U pip setuptools
Collecting pip
  Using cached https://files.pythonhosted.org/packages/c2/d7/90f34cb0d83a6c5631cf71dfe64cc1054598c843a92b400e55675cc2ac37/pip-18.1-py2.py3-none-any.whl
Collecting setuptools
  Using cached https://files.pythonhosted.org/packages/e7/16/da8cb8046149d50940c6110310983abb359bbb8cbc3539e6bef95c29428a/setuptools-40.6.2-py2.py3-none-any.whl
Installing collected packages: pip, setuptools
  Found existing installation: pip 18.0
    Uninstalling pip-18.0:
      Successfully uninstalled pip-18.0
  Found existing installation: setuptools 40.4.3
    Uninstalling setuptools-40.4.3:
      Successfully uninstalled setuptools-40.4.3
Successfully installed pip-18.1 setuptools-40.6.2
(venv) [terokinnunen@localhost tmp]$ pip install cftime
Collecting cftime
  Using cached https://files.pythonhosted.org/packages/80/23/cd5513106c709f1faa3623bb34a5284b55ea7dad4229103b68c154d8e39f/cftime-1.0.3.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-mz69t_45/cftime/setup.py", line 6, in <module>
        from Cython.Build import cythonize
    ModuleNotFoundError: No module named 'Cython'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-install-mz69t_45/cftime/
(venv) [terokinnunen@localhost tmp]$
```

**Comment by ererkka** [comment]
I confirm, same problem with cftime v. 1.0.3.
<pre>
Collecting cftime (from netcdf4>=1.3.1->-r requirements.txt (line 7))
  Using cached https://files.pythonhosted.org/packages/80/23/cd5513106c709f1faa3623bb34a5284b55ea7dad4229103b68c154d8e39f/cftime-1.0.3.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\ererkka\AppData\Local\Temp\pip-install-1vo__ug_\cftime\setup.py", line 6, in <module>
        from Cython.Build import cythonize
    ModuleNotFoundError: No module named 'Cython'
</pre>

**Comment by jswhit** [comment]
@bjlittle, can you take a look at this?

**Comment by jswhit** [comment]
Pull request #98 should fix this (again), but I haven't tested yet in an environment without cython.

**Comment by jswhit** [comment]
This was just fixed by merging pull request #98
